lib/spack/spack/spec.py

```diff
@@ -2419,6 +2426,27 @@ def from_json(stream):
                 e,
             )
 
+    @staticmethod
+    def extract_json_from_clearsig(data):
+        m = CLEARSIGN_FILE_REGEX.search(data)
+        if m:
+            return sjson.load(m.group(1))
+        return sjson.load(data)
+
+    @staticmethod
+    def from_signed_json(stream):
+        """Construct a spec from clearsigned json spec file.
+
+        Parameters:
+        stream -- string or file object to read from.
```

Comment:
```suggestion
        Args:
            stream:  string or file object to read from.
```

Reply:
Sorry, that was copy-pasted from the method above rather than the one below.  I'll fix those too.